app/src/main/java/org/oppia/android/app/devoptions/marktopicscompleted/MarkTopicsCompletedViewModel.kt

```diff
@@ -0,0 +1,78 @@
+package org.oppia.android.app.devoptions.marktopicscompleted
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.model.Topic
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.domain.devoptions.ModifyLessonProgressController
+import org.oppia.android.domain.oppialogger.OppiaLogger
+import org.oppia.android.util.data.AsyncResult
+import org.oppia.android.util.data.DataProviders.Companion.toLiveData
+import javax.inject.Inject
+
+/**
+ * [ViewModel] for [MarkTopicsCompletedFragment]. It populates the recyclerview with a list of
+ * [TopicSummaryViewModel] which in turn display the topic.
+ */
+@FragmentScope
+class MarkTopicsCompletedViewModel @Inject constructor(
+  private val oppiaLogger: OppiaLogger,
+  private val modifyLessonProgressController: ModifyLessonProgressController,
+) : ObservableViewModel() {
+
+  private var internalProfileId: Int = -1
+
+  /**
+   * List of topic ids used in [MarkTopicsCompletedFragmentPresenter] to check if all topics are
+   * selected or not.
+   */
+  val availableTopicIdList = ArrayList<String>()
```

Comment:
More Kotlin idiomatic would be to create a mutable list using ``mutableListOf<String>()``, however it's a bit concerning to expose a mutable val as part of a class's public API. Is this list directly modified by databinding? If not, we should add methods to change the list rather than exposing the list itself.

Reply:
No, it is not directly modified by the binding. I have made this private and created a getter function for it. PTAL.